zxlive/edit_panel.py

```diff
@@ -166,6 +166,14 @@ def _vert_moved(self, vs: list[tuple[VT, float, float]]) -> None:
 
     def _vert_double_clicked(self, v: VT) -> None:
         if self.graph.type(v) == VertexType.BOUNDARY:
+            input_, ok = QInputDialog.getText(
+                self, "Input Dialog", "Enter Boundary Index:"
+            )
+            try:
+                input_ = int(input_.strip())
+                self.graph.set_vdata(v, "boundary_index", input_)
```

Comment:
That makes sense. How should we get this ordered input/output information from the user in the UI?

Reply:
So there is an g.auto_detect_io() function that automatically tries to match the order of the boundary vertices to and input output list, so that could be a starting point. But then we need some kind of logic for when the user tries to change this. I'm not sure what the best way would be to do this.

Reply:
How reliable is g.auto_detect_io()? Could it lead to situations where it detects io differently on left and right sides of the rewrite?

Reply:
It does the following logic: for every boundary, check wether the thing it is connected to is to its left or to its right. If it is to its right, then it is an input, otherwise it is an output. I think it raises an exception if it is pointed vertically. Then for the ordering, it simply goes top to bottom, based on the g.get_qubit(v) property.

Reply:
That seems reasonable but it does force the user into the left to right convention, which is fine for now. Anyway, let's get rid of boundary_index and use the input/output tuple instead.

Reply:
This was actually quite straightforward. Implemented in https://github.com/Quantomatic/zxlive/pull/74/commits/eb7929f0302c18443d4b4c71b69539a143cb2b55